chunky/src/java/se/llbit/chunky/renderer/scene/camera/Camera.java

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2022 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.scene.camera;
+
+import se.llbit.chunky.renderer.scene.camera.projection.ProjectionPreset;
+import se.llbit.math.Ray;
+import se.llbit.math.Vector3;
+import se.llbit.util.JsonSerializable;
+
+import java.util.Random;
+
+public interface Camera extends JsonSerializable {
+  /**
+   * Calculate a ray shooting out of the camera based on normalized image coordinates.
+   *
+   * @param ray    result ray
+   * @param random random number stream
+   * @param x      normalized image coordinate [-0.5, 0.5]
+   * @param y      normalized image coordinate [-0.5, 0.5]
+   */
+  void calcViewRay(Ray ray, Random random, double x, double y);
+
+  /**
+   * Calculate a ray shooting out of the camera based on normalized image coordinates.
+   *
+   * @param ray result ray
+   * @param x   normalized image coordinate [-0.5, 0.5]
+   * @param y   normalized image coordinate [-0.5, 0.5]
+   */
+  void calcViewRay(Ray ray, double x, double y);
+
+  Vector3 getPosition();
+
+  double getYaw();
+  double getPitch();
+  double getRoll();
+
+  double getShiftX();
+  double getShiftY();
+
+  double getFov();
+
+  double getDof();
+  boolean isInfiniteDoF();
+  double getSubjectDistance();
+
+  ProjectionPreset getProjectionPreset();
```

Comment:
I do like this MR. The only thing I don't understand is this rename. Why call it a "projection preset"?

Reply:
My thoughts were that current projectors can be nested (e.g. parallel is a parallel projector inside a forward displacement projector inside a DoF inside a shift). It should be possible (in future plugins or PRs) to create such compositions from JSON definitions, which can be saved in the scene. At this point the name projection mode would be weird IMO, because you combine different existing projection modes to create a new projector factory. Therefore the currently provided projector factories are presets. (Does this make sense?)

Reply:
Also, CameraPresets will get refactoring treatment from me soon (and are named preset) :)

Reply:
`Projector` instead of `ProjectorPreset`? :thinking: